### PR TITLE
Arc 1545 email tech planning create email deliverability mart from sfmc tables -- PART 1

### PR DESCRIPTION
**Pull request by Frydafly**
# Pull Request for dbt-arc-functions

**This PR is 1 of 2 pull requests I expect to create for this same ticket. This Part 1 ticket resolves naming inconsistencies in our deliverability model in order to pave the way for clear naming when I add the SFMC connector**

- [x] Open the pull request as a draft so that you can run tests against the code and inspect the files in the Files view of the Pull Request interface in Github.

- [x] Once you're satisfied, convert the draft into an open Pull Request by clicking "Ready for Review" near the bottom of this page. You should not make any changes to the code except in dire emergency once you've converted from draft, so make sure you're really satisfied.

- [x] Add two of the principal contributors to this PR for review. Currently, the principal contributors are @dmbluestate , @Frydafly , and @ryantimjohn.

- [x] We've divided our PR templates into Macros, Utils and Docs PR requests. Delete the ones you're not using.
### Macros (delete if not using)

- [x] Link Jira tickets (or Github Issues) to PR here. Make sure that the Jira ticket has a good description as to why this PR is necessary and the business problem that this PR is solving. Moreover, if there have been any discussions of this ticket in Slack, make sure they're linked in the Jira ticket:
https://bluestate.atlassian.net/browse/ARC-1545?atlOrigin=eyJpIjoiM2ZmNzQwNjAwZWY1NGU1N2JhMjY0YjI0NGMxMzY3NjAiLCJwIjoiaiJ9 

- [x] Make sure this is named using Jira extension so that naming convention is consistent. If it's not, that's okay. Go to the Jira ticket, click into the ticket, then on the right under Details, find Development, then click the Create branch button. Copy the Branch Name from there. Read here [how to rename a branch in Github](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) to rename this branch. In the future, create branches from Jira.

- [x] Make sure the Macro is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.
**Habitat for Humanity**
branch: https://github.com/bsd/bsd-habitat/tree/test_arc-1541_email_deliverability 

- [x] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page

Screenshot 2023-08-08 at 11 40 34 AM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this
Screenshot 2023-08-08 at 11 43 07 AM

## Formatters

Three different formatters will run when you open a pull request:
- [sqlfmt](http://sqlfmt.com/)
- [yamlfix](https://lyz-code.github.io/yamlfix/)
- [Black(python)](https://pypi.org/project/black/)

If any of them detect syntax problems, they will error. They will automatically open a pull request with the changes you need to make to pass the formatter. Just merge that PR and you should be good to go.

## Linters
We have two in-house linters:

### [check_macros.py](https://github.com/bsd/dbt-arc-functions/blob/main/utils/check_macros.py)
Makes sure that every macro has a doc associated with it in the documentation folder with the same folder structure, i.e.:

`documentation/combined_email_paidmedia/marts/mart_combined_email_paidmedia_daily_revenue_performance.yml`

is the doc for

`       macros/combined_email_paidmedia/marts/mart_combined_email_paidmedia_daily_revenue_performance.sql`

Additionally makes sure that macros are named correctly, i.e. with the naming convention `create_(macro_file_name_stem)`

### [check_docs_correct.py](https://github.com/bsd/dbt-arc-functions/blob/main/utils/check_docs_correct.py)
Runs through a lot of checks on documents to make sure that they are formatted to our standards. This includes checking whether docs and sources:
- have columns
- have tables
- source columns have data_type and description
- have a version number
- docs have a macro associated with them
- docs are not blank
- that model names match the filename stem

There may be additional checks not documented here. The linter will fail on any error.

The script will document all failures and display them to you with remedies.

### CODE REVIEW

**Comment by Frydafly** [comment]
Hello! @ryantimjohn @dmbluestate please also review how these changes are implemented in the bsd-habitat project, so that I may merge updates to habitat as well

https://github.com/bsd/bsd-habitat/pull/125

**Review by ryantimjohn — APPROVED** [review]

**Comment by ryantimjohn** [comment]
@Frydafly we'll need to bump a full version on dbt-arc-functions as this will break previous versions because of name change

**Comment by Frydafly** [comment]
@ryantimjohn sounds good -- even if it only breaks habitat and I already have a fix for that? let me know what's the appropriate thing to do

**Comment by Frydafly** [comment]
@ryantimjohn I'd like to adhere, yes. I'll merge this, create release that bumps full version, edit PR in habitat to have revision: v7.0.0 and then merge that one. 

Let me know if I'm getting the order or process wrong.

**Comment by ryantimjohn** [comment]
looks great!

File: macros/email_deliverability/staging/stg_email_deliverability_clicks_daily_rollup_unioned.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
-{% macro create_stg_email_deliverability_person_bounces_daily_rollup_unioned() %}
+{% macro create_stg_email_deliverability_clicks_daily_rollup_unioned() %}
     {% set relations = dbt_arc_functions.relations_that_match_regex(
-        "^stg_.*_person_bounces_daily_rollup$"
+        "^stg_.*_deliverability_clicks_daily_rollup$"
```
[the remark this answers is not in the document]
Comment (dmbluestate) [reply macros/email_deliverability/staging/stg_email_deliverability_clicks_daily_rollup_unioned.sql:3]: no worries. After reviewing the other changes, I understand what happened. Thanks!

**Review by dmbluestate — APPROVED** [review]

**Pull request merged**